finagle-core/src/main/scala/com/twitter/finagle/service/RateLimitingFilter.scala

```diff
@@ -11,22 +11,25 @@ import com.twitter.finagle.{RefusedByRateLimiter, Service, SimpleFilter}
 class LocalRateLimitingStrategy[Req](categorizer: Req => String, windowSize: Duration, rate: Int)
     extends (Req => Future[Boolean]) {
 
+  require(rate > 0, s"rate $rate must be strictly positive.")
+
   private[this] val rates = mutable.HashMap.empty[String, (Int, Time)]
 
-  def apply(req: Req) = synchronized {
+  def apply(req: Req): Future[Boolean] = synchronized {
     val now = Time.now
     val id = categorizer(req)
     val (remainingRequests, timestamp) = rates.getOrElse(id, (rate, now))
 
     val accept = if (timestamp.until(now) > windowSize) {
-      rates(id) = (rate, now)
+      rates(id) = (rate - 1, now)
```

Comment:
Why did this change?

Reply:
I added a test for that, else counter is not well reset after one elapsed window time.

Reply:
I removed:
```
if (rate <= 0) {
      false
```
Because now rate is necessarily strictly positive at construction time (with require condition rate <= 0 is impossible). 
`remainingRequests` resolved case of no more remaining request.

Reply:
I added a test for that: 
```
 rates(id) = (rate - 1, now)
```
Else counter is not well reset after a elapsed time window

Reply:
Do you agree with these changes ?

Reply:
Sorry missed notifications about this.

No, I do not agree. This is clearly a behavior change and I'm not sure why to have to do that. Is in optimization? Or code clean up?

Reply:
It was about debut your rate limiter. As you want, I removed that and test in relation to this change. I let only test if rate limiter is negative.